python/src/aiconfig/editor/client/src/components/AIConfigEditor.tsx

```diff
@@ -1038,16 +1038,18 @@ function AIConfigEditorBase({
     }
 
     const interval = setInterval(async () => {
-      try {
-        const res = await getServerStatusCallback();
-        setServerStatus(res.status);
-      } catch (err: unknown) {
-        setServerStatus("ERROR");
+      if (!readOnly) {
```

Comment:
This can just be done above with the other checks to return, rather than setting the interval (otherwise it will unnecessarily run this interval every heartbeat_interval ms

Reply:
If `readOnly` is in the dependency array for the effect then the effect will rerun and set a new interval (and dispose of the existing one in the case of toggling *to* readonly)

Reply:
thanks for clarifying, updated as suggested.